bin/build_update_subfolders.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+
+import send_email
+import synapseclient
+import synapseutils
+import sys
+
+
+def is_new_submitter(syn, submitter_id, parent_folder_id):
+    """
+    Used to check if the submitter is new or already has an existing folder.
+
+    This function uses ``synapseutils.walk`` to crawl through the level 1 subfolders
+    (i.e. The folders 1 level below the Parent-Folder, that are named after the submitter's synapse ID) to
+    verify whether or not a submitter already has an existing folder created.
+    """
+
+    submitter_subfolders = []
+    # Setting ``include_types`` to only Folders so that Files are excluded, and walk is faster.
+    for _, level1_subfolders, _ in synapseutils.walk(syn=syn, synId=parent_folder_id,
+                                                     includeTypes=["folder"]
+                                                     ):
+        for subfolder in level1_subfolders:
+            # ``folder`` is a tuple structured as follows: (Folder name, synID). Let's just get the Folder name.
+            submitter_subfolders.append(subfolder[0])
+
+    if submitter_id in submitter_subfolders:
+        return False
+
+    return True
```

Comment:
If I understand this function correctly, I think we can further optimize it by returning `False` as default.  Then, in the logic above, iterate through the subfolders and return `True` as soon as the user Id is found, rather than append the subfolders into a list first and checking for `submitter_id` afterward.

Reply:
Resolving this since we are removing this function per conversation [here](https://github.com/Sage-Bionetworks-Workflows/nf-synapse-challenge/pull/5#discussion_r1480414505)